feature/subscriber-attributes/src/main/java/com/revenuecat/purchases/subscriberattributes/SubscriberAttributesManager.kt

```diff
@@ -106,4 +116,57 @@ class SubscriberAttributesManager(
 
         deviceCache.setAttributes(appUserID, attributesToBeSet)
     }
+
+    /**
+     * Collect gpsAdId and androidId automatically
+     *
+     * This method cannot be called in the main thread as it may block leading to ANRs.
+     * An IllegalStateException will be thrown if this is called on the main thread.
+     */
+    fun collectDeviceIdentifiers(
+        appUserID: String,
+        applicationContext: Application
+    ) {
+        val attributesToSet = getDeviceIdentifiers(applicationContext)
+        setAttributes(attributesToSet, appUserID)
+    }
+
+    private fun getDeviceIdentifiers(applicationContext: Application): MutableMap<String, String?> {
+        val deviceIdentifiers = mutableMapOf<String, String?>()
+        try {
+            val adInfo = AdvertisingIdClient.getAdvertisingIdInfo(applicationContext)
+            if (!adInfo.isLimitAdTrackingEnabled) {
+                val advertisingID = adInfo.id
+                deviceIdentifiers[SubscriberAttributeKey.DeviceIdentifiers.GPSAdID.backendKey] = advertisingID
+            }
+        } catch (e: GooglePlayServicesNotAvailableException) {
+            errorLog("GooglePlayServices is not installed. Couldn't get and advertising identifier. " +
+                "Message: ${e.localizedMessage}")
+        } catch (e: GooglePlayServicesRepairableException) {
+            errorLog("GooglePlayServicesRepairableException when getting advertising identifier. " +
+                "Message: ${e.localizedMessage}")
+        }
+        @SuppressLint("HardwareIds")
+        val androidID =
+            Settings.Secure.getString(applicationContext.contentResolver, Settings.Secure.ANDROID_ID)
+        deviceIdentifiers[SubscriberAttributeKey.DeviceIdentifiers.AndroidID.backendKey] = androidID
+        return deviceIdentifiers
+    }
+
+    /**
+     * Set the specific ID for the specified attribution network. It also collects GPS ID and ANDROID_ID automatically.
+     *
+     * This method cannot be called in the main thread as it may block leading to ANRs.
+     * An IllegalStateException will be thrown if this is called on the main thread.
+     */
```

Comment:
same thought here

Reply:
If I recall correctly, the "reason" why I did the change to background in Purchases is because we are using the same ExecutorService as the HttpClient, and we need to synchronize the access to the ExecutorService and we already have the logic in Purchases. I agree this should be outside of Purchases and that's what I did first place. Let me rethink of a better solution than moving the change to the background to Purchases.